### ISSUE TITLE
Missing meta_query when querying for orders to sync

### ISSUE THREAD

**Issue by Saggre** [PostiDigital/woocommerce-warehouse-plugin#13]
Regarding [this change from `get_posts` to `wc_get_orders`.](https://github.com/PostiDigital/woocommerce-warehouse-plugin/commit/ea37c33a0f9be8954e77ead67256d1172f12ca54#diff-f05a6f993ddf8ee7ff9cd31edc30f4894bc17ea4cdbcdd911210ab54664748c7L311-R311)

I think WooCommerce ignores the `meta_query` arg when parsing arguments directly provided to `wc_get_orders`.
It's ignored at [woocommerce/includes/data-stores/class-wc-data-store-wp.php:284](https://plugin.dog/plugin/woocommerce/9.2.3?path=/includes/data-stores/class-wc-data-store-wp.php#L284)

WooCommerce ends up building the following query where meta_query for posti order id is missing (transformed to json):

```json
{
  "errors": [],
  "meta_query": [],
  "posts_per_page": -1,
  "paginate": false,
  "order": "DESC",
  "orderby": "date",
  "fields": "objects",
  "post_type": "shop_order",
  "post_status": "any",
  "numberposts": -1,
  "paged": 1,
  "date_query": [],
  "no_found_rows": true
}
```

I tested this with the following hook:

```php
add_filter( 'woocommerce_order_data_store_cpt_get_orders_query', function ( $wp_query_args, $query_vars ) {
    // Check the variables.
    return $wp_query_args;
}, 10, 2 );
```

Let me know if I can provide more info.

**Comment by suhhomi1** [comment]
Hi, thank you for looking into this but I'm unable to confirm the problem.
For example during order status sync the '_posti_id' field is actually being applied on WP 6.6.1 + WC 9.1.2,

**Comment by Saggre** [comment]
@suhhomi1 It seems to depend on whether HPOS is being used.

If HPOS is being used, the order query is managed by the newer `OrdersTableDataStore` class. If HPOS is not being used, the order query is managed by the old `WC_Order_Data_Store_CPT` class.

You can check which one is being used with this: `WC_Data_Store::load( 'order' )->get_current_class_name();`

---

Here's a minimal illustration of the problem:

1. Set up a store with WP 6.6.2 and WC 9.2.3 (without woocommerce-warehouse-plugin)
2. Disable HPOS
3. Add the following mu-plugin

```php
<?php

add_filter( 'woocommerce_order_data_store_cpt_get_orders_query', function ( $query_args ) {
    echo '<pre>';
    var_dump( $query_args );
    echo '<pre>';

    return $query_args;
} );

add_action( 'wp_head', function() {
    $orders = wc_get_orders( array(
        'post_type' => 'shop_order',
        'post_status' => 'any',
        'numberposts' => -1,
        'meta_query' => array(
        'relation' => 'AND',
            array(
                'key' => '_posti_id',
                'value' => array( '123' ),
                'compare' => 'IN'
            ),
        ),
    ) );

    exit;
} );
```

4. Load WP frontend. Result with the meta query missing:

```text
array(13) {
  ["errors"]=>
  array(0) {
  }
  ["meta_query"]=>
  array(0) {
  }
  ["posts_per_page"]=>
  int(-1)
  ["paginate"]=>
  bool(false)
  ["order"]=>
  string(4) "DESC"
  ["orderby"]=>
  string(4) "date"
  ["fields"]=>
  string(7) "objects"
  ["post_type"]=>
  string(10) "shop_order"
  ["post_status"]=>
  string(3) "any"
  ["numberposts"]=>
  int(-1)
  ["paged"]=>
  int(1)
  ["date_query"]=>
  array(0) {
  }
  ["no_found_rows"]=>
  bool(true)
}
array(13) {
  ["errors"]=>
  array(0) {
  }
  ["meta_query"]=>
  array(0) {
  }
  ["posts_per_page"]=>
  int(-1)
  ["paginate"]=>
  bool(false)
  ["order"]=>
  string(4) "DESC"
  ["orderby"]=>
  string(4) "date"
  ["fields"]=>
  string(7) "objects"
  ["post_type"]=>
  string(17) "shop_order_refund"
  ["post_parent__in"]=>
  array(1) {
    [0]=>
    int(14)
  }
  ["post_status"]=>
  array(8) {
    [0]=>
    string(10) "wc-pending"
    [1]=>
    string(13) "wc-processing"
    [2]=>
    string(10) "wc-on-hold"
    [3]=>
    string(12) "wc-completed"
    [4]=>
    string(12) "wc-cancelled"
    [5]=>
    string(11) "wc-refunded"
    [6]=>
    string(9) "wc-failed"
    [7]=>
    &string(17) "wc-checkout-draft"
  }
  ["paged"]=>
  int(1)
  ["date_query"]=>
  array(0) {
  }
  ["no_found_rows"]=>
  bool(true)
}
```

Note that with HPOS enabled the hook used for debugging will not be called.

**Comment by suhhomi1** [comment]
@Saggre please try version 3.0.1, I added a workaround for non-HPOS case.

**Comment by Saggre** [comment]
@suhhomi1 Thanks. I'm yet to test the plugin, but the change seems sound.

I'm perfectly happy with the update, but if you want to avoid the workaround, I think these untested args will result in the same query in all cases:

```php
$posts_query = array(
	'post_type' => 'shop_order',
	'post_status' => 'any',
	'numberposts' => -1,
	'meta_key'     => '_posti_id',
	'meta_value'   => $order_ids,
	'meta_compare' => 'IN',
);
```

**Comment by suhhomi1** [comment]
Thank you. I've applied your suggestion instead of workaround and it is now released as 3.0.1.

**Comment by Saggre** [comment]
The issue was resolved by this change
